### PR TITLE
linux: enable PERSISTENT_KEYRINGS and KEYS_REQUEST_CACHE

### PR DESCRIPTION
**Pull request by NickCao**
PERSISTENT_KEYRINGS provides a register of persistent per-UID keyrings, useful for encrypting storage pools in stratis.
KEYS_REQUEST_CACHE enable temporary caching of the last request_key() result.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by NickCao** [comment]
Result of `nixpkgs-review pr 184770` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>linux_testing_bcachefs</li>
  </ul>
</details>
<details>
  <summary>3 packages failed to build:</summary>
  <ul>
    <li>linux_5_4_hardened</li>
    <li>linux_latest-libre</li>
    <li>linux_mptcp</li>
  </ul>
</details>
<details>
  <summary>9 packages built:</summary>
  <ul>
    <li>linux_4_14_hardened</li>
    <li>linux_4_19_hardened</li>
    <li>linux_5_10_hardened</li>
    <li>linux_hardened (linux_5_15_hardened)</li>
    <li>linux_5_18_hardened</li>
    <li>linux_latest (linux_testing)</li>
    <li>linux_xanmod</li>
    <li>linux_xanmod_latest</li>
    <li>linux_xanmod_tt</li>
  </ul>
</details>

Failures seem unrelated.

File: pkgs/os-specific/linux/kernel/common-config.nix [hunk 1]
```diff
@@ -497,6 +497,9 @@ let
       # Depends on MODULE_SIG and only really helps when you sign your modules
       # and enforce signatures which we don't do by default.
       SECURITY_LOCKDOWN_LSM = option no;
+
+      PERSISTENT_KEYRINGS              = yes;
+      KEYS_REQUEST_CACHE               = whenAtLeast "5.3" yes;
```
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/kernel/common-config.nix:502]:
```suggestion
      # provides a register of persistent per-UID keyrings, useful for encrypting storage pools in stratis
      PERSISTENT_KEYRINGS              = yes;
      # enable temporary caching of the last request_key() result
      KEYS_REQUEST_CACHE               = whenAtLeast "5.3" yes;
```

**Comment by lovesegfault** [comment]
Can you please provide some info on whether other distros enable this by default? In the past, we've looked at the default Debian, Fedora, and Arch kernel configs to back-up the request of new enabled configs.

**Comment by NickCao** [comment]
Arch Linux: https://github.com/archlinux/svntogit-packages/blob/230ed05b6ffabad326468e605affabb904b25e01/trunk/config#L10232-L10233
Fedora: https://src.fedoraproject.org/rpms/kernel/blob/rawhide/f/kernel-x86_64-fedora.config#_4830

**Pull request merged**